main.go

```diff
@@ -1380,6 +1390,16 @@ func listen(l, controlListener net.Listener, err error) {
 				"prefix": "main",
 			}).Info("Custom gateway started")
 			go s.Serve(l)
+
+			if controlListener != nil {
+				cs := &http.Server{
+					ReadTimeout:  time.Duration(readTimeout) * time.Second,
+					WriteTimeout: time.Duration(writeTimeout) * time.Second,
+					Handler:      controlRouter,
+				}
+				go cs.Serve(controlListener)
+			}
+
```

Comment:
Yeah, we can do that in future, this block does have differences, but minor ones. Let's make it as separate PR, maybe you will touch this part someday :)